File: Leetcode/3042. Count Prefix and Suffix Pairs I.py
class Solution:
    def countPrefixSuffixPairs(self, words: List[str]) -> int:
        def isPrefixAndSuffix(str1,str2):
            return str2.find(str1)==0 and str2.find(str1,len(str2)-len(str1))!=-1
        n=len(words)
        res=0
        for i in range(0,n):
            for j in range(i+1,n):
                if isPrefixAndSuffix(words[i],words[j]):
                    res+=1
        return res

File: Leetcode/3042. Count Prefix and Suffix Pairs I.cpp
class Solution {
public:
    bool isPrefixAndSuffix(string& str1,string& str2){
        return str2.find(str1)==0 && str2.find(str1,str2.size()-str1.size())!=string::npos;
    }

    int countPrefixSuffixPairs(vector<string>& words) {
        int n=words.size();
        int res=0;
        for(int i=0;i<n;i++){
            for(int j=i+1;j<n;j++){
                if(isPrefixAndSuffix(words[i],words[j]))    res++;
            }
        }
        return res;
    }
};
